app/models/watchlist_instance.rb

```diff
@@ -113,62 +195,67 @@ def resolve_watchlist_instance
   def self.add_new_instances_for_conditions(conditions, course)
     new_instances = []
     course_user_data = CourseUserDatum.where(course_id: course.id, instructor: false, course_assistant: false)
-    # HUGE TRANSACTION AHEAD
-    ActiveRecord::Base.transaction do
-      conditions.each do |condition|
-        case condition.condition_type
+    
+    conditions.each do |condition|
+      case condition.condition_type
+      
+      when "grace_day_usage"
+        grace_day_threshold = condition.parameters[:grace_day_threshold].to_i
+        date = condition.parameters[:date]
+        asmts_before_date = course.asmts_before_date(date)
+        next if asmts_before_date.count == 0 # go to the next condition loop if there is no latest assessment
```

Comment:
I saw that there are some changes here, could you let me know in general what this change is about?

Reply:
Previously, the 'next' is inside of the loop for each course_user_datum when I fail to acquire their corresponding latest aud before date. I thought it would be better if I pull it out and check this condition on an assessment level and stop a little early if there are no assessments at all before the specified date.